WrightTools/artists/_interact.py

```diff
@@ -331,8 +326,7 @@ def update(info):
             x0 = info.xdata
             y0 = info.ydata
             if x0 is None or y0 is None:
-                print(info, info.xydata)
-                raise AttributeError
+                raise AttributeError(info)
```

Comment:
Commented this on a now outdated version, repeating for visibility:

I would say  TypeError

AttributeError tends to specifically mean the call to __getattr__ (which is called by doing foo.bar) fails. That is not the case here, as the attribute of info is defined (or at least does not raise an exception)
(Python docs)[https://docs.python.org/3.7/library/exceptions.html#AttributeError]

TypeError is appropriate as x0/y0 have NoneType, not the type expected (I beleive numeric, maybe float specifically). We define in exceptions.py a TypeError which inherits from both our own base exception and the standard library TypeError, so it can be caught by a calling function by either.

ValueError is a similar catchall, but is defined as being the correct type, which I do not think applies here.

I am not opposed to adding AttributeError to exceptions.py. Mostly the Idea is to have a base exception class from which all exceptions thrown by WrightTools itself can be caught. We did not define all of the standard lib exception, as we do not anticipate needing most of them.

Admittedly for the current scope of where interact2D is anticipated to be used, this kind of boxing of exceptions seems slightly superfluous. However, I'd like to have these systems in place and make sure it can be called in some larger application as well (perhaps as a pane in a larger GUI app)